influence_benchmark/stats/plot_retroactive_evals.py

```diff
@@ -0,0 +1,256 @@
+        else:
+            print(f"Warning: Results file not found for iteration {iteration_number}")
+
+    return results_dfs
+
+
+def get_extreme_entries(results_dfs: List[pd.DataFrame], metric: str, n: int = 5) -> Dict[str, pd.DataFrame]:
+    # Concatenate all DataFrames
+    all_results = pd.concat(results_dfs, ignore_index=True)
+
+    # Sort the DataFrame by the specified metric
+    sorted_df = all_results.sort_values(by=metric, ascending=False)
+
+    # Get the highest and lowest n entries
+    highest = sorted_df.head(n)
+    lowest = sorted_df.tail(n)
+
+    return {"highest": highest, "lowest": lowest}
+
+
+def format_conversation(conversation: List[Dict[str, str]]) -> str:
+    formatted_output = ""
+    for turn in conversation:
+        role = turn["role"].capitalize()
+        content = turn["content"].strip()
+
+        formatted_output += f"{role}:\n"
+        formatted_output += f"{content}\n\n"
+
+    return formatted_output.strip()
+
+
+def evaluate_iteration(
+    run_name,
+    iteration_number,
+    backend_config,
+    eval_config,
+    metrics,
+    batch_size,
+    gpu_ids,
+    save=True,
+    env_name_prefix="",
+    env_config_path=PROJECT_ROOT / "config/env_configs/normal_therapist",
+):
+    print(f"Processing iteration {iteration_number}")
+    iteration_path = PROJECT_DATA / "trajectories" / run_name / str(iteration_number)
+
+    evaluator = RetroactiveIterationEvaluator(
+        data=iteration_path,
+        backend_config=backend_config,
+        config={"preference": eval_config["preference"]},
+        metrics=metrics,
+        batch_size=batch_size,
+        devices=gpu_ids,
+        env_name_prefix=env_name_prefix,
+        env_config_path=env_config_path,
+    )
+
+    results_df = evaluator.evaluate_iteration()
+
+    print("Evaluation completed.")
+
+    if save:
+        # Save the results
+        output_path = (
+            PROJECT_DATA / "trajectories" / run_name / f"{iteration_number}_retro_pref" / "retroactive_eval.json"
+        )
+        output_path.parent.mkdir(parents=True, exist_ok=True)
+        results_df.to_json(output_path, orient="records")
+        print(f"Results saved to: {output_path}")
+
+    return results_df
+
+
+if __name__ == "__main__":
+    # Load configuration from retro_single.yaml
+    config_path = PROJECT_ROOT / "config/experiment_configs/retroactive_evals/retro_single.yaml"
+    config = load_yaml(str(config_path))
+
+    # Extract configuration values
+    eval_config = load_yaml(str(PROJECT_ROOT / config["eval_config"]))
+    backend_config = config["backend_config"]
+    run_name = config["run_name"]
+    metrics = config["metrics"]
+    batch_size = config["batch_size"]
+    gpu_ids = config["gpu_ids"]
+    iteration_number = config["iteration_number"]
+    save = config["save"]
+    env_name_prefix = config.get("env_name_prefix", "")
+    env_config_path = PROJECT_ROOT / config.get("env_config_path", "config/env_configs/normal_therapist")
```

Comment:
We currently have these lines repeated also in the notebook 2 times I think

Reply:
I think for transparency it's better to have the constructor of `RetroactiveIterationEvaluator` spell out exactly the inputs it is expecting rather than taking in a dict. But now that we do not have a config, I just set these parameters in python in the script/notebook so it should be fine.

It was repeated in the notebook two times because it was loading 2 different config files -- one for a single iteration and the other for all iterations in a run. But now that is set in python too, so it's not repeated.